rmw_fastrtps_shared_cpp/src/rmw_response.cpp

```diff
@@ -62,8 +62,9 @@ __rmw_take_response(
     info->response_type_support_->deserializeROSmessage(
       deser, ros_response, info->response_type_support_impl_);
 
-    request_header->source_timestamp = response.sample_info_.sourceTimestamp.to_ns();
-    request_header->received_timestamp = response.sample_info_.receptionTimestamp.to_ns();
+    request_header->source_timestamp = response.sample_info_.source_timestamp.to_ns();
```

Comment:
Please, remove header file `fastrtps/subscriber/Subscriber.h` which is not being used, and instead add the header file `fastdds/rtps/common/WriteParams.h` which is indeed being used.